scratch.py

```diff
@@ -115,80 +117,96 @@ def pack_string(string):
                     incident.pilot_total_hours,
                     incident.midair
                 ))
-            fh.write(data)
+
+            file.write(data)
         except Exception as err:
             print(err)
+
         return True
 
     def import_binary(self, filename):
-        print("4343")
-        NumbersStruct = struct.Struct("<Idi?")
-        def unpack_string(fh, eof_is_error=True):
-            print(eof_is_error)
+        print("Entering `import_binary`")
+
+        def unpack_string(file, eof_is_error=False):
+            print(f"Entering `unpack_string`... {eof_is_error= }")
             uint16 = struct.Struct("<H")
-            length_data = fh.read(uint16.size)
-            print("yes", length_data,87)
+            length_data = file.read(uint16.size)
+            print(f"{length_data= }")
 
             if not length_data:
                 if eof_is_error:
                     raise ValueError("missing or corrupt string size")
-                return 78
+                return None
 
             length = uint16.unpack(length_data)[0]
-            print(length, "2343")
-            if length == 0:
+            print(f"{length= }")
+            if not length:
                 return ""
-            data = fh.read(length)
-            print(data,"re")
-            print(data,"me",len(data),length,len(data) != length)
-            if not data or len(data) != length:
+
+            data = file.read(length)
+            print(f"{data= }")
+            print(data, len(data), length)
+            if len(data) != length:
                 raise ValueError("missing or corrupt string")
-            format = "<{0}s".format(length)
-            return struct.unpack(format, data)[0].decode("utf8")
 
-        fh = None
+            return struct.unpack(f"<{length}s", data)[0].decode("utf8")
+
+
+        NumbersStruct = struct.Struct("<Idi?")
+        file = None
         MAGIC = b"AIB\x00"
         FORMAT_VERSION = b"\x00\x01"
         GZIP_MAGIC = b"\x1F\x8B"
-        try:
-            fh = open(filename, "rb")
-            magic = fh.read(len(GZIP_MAGIC))
-            if magic == GZIP_MAGIC:
-                fh.close()
-                fh = gzip.open(filename, "rb")
-            else:
-                fh.seek(0)
-            magic = fh.read(len(MAGIC))
-            if magic != MAGIC:
-                raise ValueError("invalid .aib file format")
-            version = fh.read(len(FORMAT_VERSION))
-            if version > FORMAT_VERSION:
-                raise ValueError("unrecognized .aib file version")
-            self.clear()
-        except Exception as err:
-            print(err)
```

Comment:
The errors shouldn't be handled since they are potential causes of other errors.